### ISSUE TITLE
Can't statically link mingw's libcurl

### ISSUE THREAD

**Issue by sitsofe** [msys2/MINGW-packages#4404]
I installed cygwin's mingw-w64-curl but I get the following:
```
cat > curltest.c <<EOF
#include <curl/curl.h>

void main() {
    curl_easy_init();
}
EOF
 x86_64-w64-mingw32-gcc curltest.c -static $(/usr/x86_64-w64-mingw32/sys-root/mingw/bin/curl-config --static-libs)
```
produces
```
/cygdrive/c/cygwin64/bin/x86_64-w64-mingw32-ld: cannot find -lnghttp2
collect2: error: ld returned 1 exit status
```
So it looks like to static link curl all of curl's dependencies must have static libraries available too. For the record running
```
/usr/x86_64-w64-mingw32/sys-root/mingw/bin/curl-config --static-libs
```
produces:
```
/usr/x86_64-w64-mingw32/sys-root/mingw/lib/libcurl.a -lnghttp2 -lnormaliz -lssh2 -lssh2 -lws2_32 -lbcrypt -lcrypt32 -lws2_32 -lbcrypt -lcrypt32 -lz -lssl -lcrypto -lgdi32 -lcrypt32 -lwldap32 -lz -lws2_32
```

(this is a follow on from issue #950)

**Comment by muv1** [comment]
You can try this:

~~~
cc a.c \
-Wl,--start-group \
-lcrypt32 \
-lcurl \
-lwldap32 \
-lws2_32 \
-lz \
-Wl,--end-group \
-DCURL_STATICLIB \
-static
~~~

<https://github.com/muv1/gulf/blob/51137955/curl-branch/program.sh>

**Comment by sitsofe** [comment]
As hinted at by @muv1, on Windows you actually need to define at least two defines:
- `CURL_STATICLIB` so curl's includes know what you're attempting
- `NGHTTP2_STATICLIB` so nghttp2's includes know what you're attempting

This is thoroughly described in https://github.com/curl/curl/issues/2516#issuecomment-383467678 (thanks to @muv1 for pointing this out). Given everything is working as intended and the user just needs to know the correct incantations I'm closing this.

**Comment by muv1** [comment]
I have deleted my previous comment, as its not currently possible to statically link with MSYS2 LibCurl. The package version is broken. Alternative is to build LibCurl yourself, or use the LibCurl from here:

<https://curl.haxx.se/windows>

note that only LibCurl package itself is broken, the other MSYS2 packages are fine. See here for more info:

<https://github.com/muv1/gulf>

**Comment by MehdiChinoune** [comment]
Fixed by #8804
